contracts/access/manager/AccessManaged.sol

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+import {IAuthority} from "./IAuthority.sol";
+import {IManaged} from "./IManaged.sol";
+import {Context} from "../../utils/Context.sol";
+
+abstract contract AccessManaged is Context, IManaged {
+    address private _authority;
+
+    constructor(address initialAuthority) {
+        _setAuthority(initialAuthority);
+    }
+
+    modifier restricted() {
+        _checkCanCall(_msgSender(), address(this), msg.sig);
+        _;
+    }
+
+    function authority() public view virtual returns (address) {
+        return _authority;
+    }
+
+    function updateAuthority(address newAuthority) public virtual {
```

Comment:
The name should be [`setAuthority`](https://github.com/dapphub/ds-auth/blob/master/src/auth.sol#L45) to match the existing interface.